network/tls_authentication.go

```diff
@@ -102,6 +103,36 @@ func (c hexIdentifierConverter) Convert(identifier []byte) ([]byte, error) {
 	return out, nil
 }
 
+// TLSClientIDExtractor complex component for extracting clientID from certificates using certificate identifier extractor and convertor
+type TLSClientIDExtractor interface {
+	ExtractClientID(certificate *x509.Certificate) ([]byte, error)
+}
```

Comment:
I'm happy to see these two things merged. Though, this capitalization is kinda confusing to read: TLS—Client—IDE—xtractor. I constantly have to correct myself that this has nothing to do with IDEs.

I do not have any alternatives to suggest, unfortunately.